models/spring-ai-azure-openai/src/main/java/org/springframework/ai/azure/openai/AzureOpenAiChatClient.java

```diff
@@ -443,8 +442,11 @@ protected ChatCompletionsOptions doCreateToolResponseRequest(ChatCompletionsOpti
 
 			String functionResponse = this.functionCallbackRegister.get(functionName).call(functionArguments);
 
-			// Add the function response to the conversation.
-			conversationHistory.add(new ChatRequestToolMessage(functionResponse, toolCall.getId()));
+			if (functionResponse != null) {
```

Comment:
I believe that the CompleteRoundTripBox and the runtime detection of needCompleteRoundTrip are unnecessary.
Since in the FunctionCallbackContext you have already detected Void response (and Consumer)  functions it should be possible add new `boolean noReturnFunction()` method to the `FunctionCallback` interface. Then in the  AbstractFunctionCallSupport#handleFunctionCallOrReturn you should be able to use this to decide how to handle the return? 
This way you can implement everything inside the abstract support classes without changing the Model implementations?

Reply:
yes but in model implemetation we "extract" the function to call and call it. If the model support more than one function or we change the implementation to extract all involved functions and in AbstractFunctionCallback we decide if a second roundtrip is needed... 

Try to clean model implementation :thinking: